bulletroboy/exoforce/exoforce_hardware.py

```diff
@@ -0,0 +1,266 @@
+import numpy as np
+from pyquaternion import Quaternion
+
+from .exoforce import ExoForce
+from ..utils.utils import Topics, Services
+
+from roboy_simulation_msgs.msg import Collision
+from roboy_middleware_msgs.msg import MotorCommand, MotorState
+from roboy_middleware_msgs.srv import ControlMode
+from geometry_msgs.msg import PoseStamped
+
+from rclpy.callback_groups import ReentrantCallbackGroup
+
+FAST_PWM_TENSION = 300
+MIN_PWM_TENSION = 150
+YES_OPTIONS = ["Yes","yes","y",""]
+
+POS_CONTROL_THRESHOLD = 0.1
+
+class ExoforceHW(ExoForce):
+	def __init__(self, cage_conf):
+		"""
+		Args:
+			cage_conf (CageConfiguration): Cage configuration defined in the configuration file.
+		
+		"""
+		super().__init__(cage_conf, "exoforce", POS_CONTROL_THRESHOLD)
+		self.callback_group = ReentrantCallbackGroup()
+
+		self.create_subscription(Collision, Topics.MAPPED_COLLISIONS, self.collision_listener, 1)
+		self.create_subscription(Collision, 'roboy/simulation/roboy/collision_hw', self.collision_listener, 1)
```

Comment:
if you are replacing topic names, why not replace everything?

Reply:
This subscription is just a workaround and will be deleted when the force mapping is fixed

Reply:
see https://github.com/Roboy/bulletroboy/pull/25#issuecomment-760128816